tests/oc_template_test.go

```diff
@@ -0,0 +1,331 @@
+					\"template\": {
+						\"spec\": {
+							\"domain\": {
+								\"resources\": {
+									\"requests\": {
+										\"memory\": \"${{MEMORY}}\"
+									}
+								},
+								\"cpu\": {
+									\"cores\": \"${{CPU_CORES}}\"
+								},
+								\"devices\": {
+									\"disks\": [
+										{
+											\"volumeName\": \"registryvolume\",
+											\"disk\": {
+												\"bus\": \"virtio\"
+											},
+											\"name\": \"disk0\"
+										},
+										{
+											\"volumeName\": \"cloudinitvolume\",
+											\"disk\": {
+												\"bus\": \"virtio\"
+											},
+											\"name\": \"disk1\"
+										}
+									]
+								}
+							},
+							\"volumes\": [
+								{
+									\"name\": \"registryvolume\",
+									\"registryDisk\": {
+										\"image\": \"kubevirt/fedora-cloud-registry-disk-demo:latest\"
+									}
+								},
+								{
+									\"name\": \"cloudinitvolume\",
+									\"cloudInitNoCloud\": {
+										\"userDataBase64\": \"I2Nsb3VkLWNvbmZpZwpwYXNzd29yZDogYXRvbWljCnNzaF9wd2F1dGg6IFRydWUKY2hwYXNzd2Q6IHsgZXhwaXJlOiBGYWxzZSB9Cg==\"
+									}
+								}
+							]
+						},
+						\"metadata\": {
+							\"labels\": {
+								\"kubevirt-ovm\": \"ovm-${NAME}\"
+							}
+						}
+					}
+				},
+				\"apiVersion\": \"kubevirt.io/v1alpha1\",
+				\"metadata\": {
+					\"labels\": {
+						\"kubevirt-ovm\": \"ovm-${NAME}\"
+					},
+					\"name\": \"${NAME}\"
+				}
+			}
+		],
+		\"apiVersion\": \"v1\",
+		\"parameters\": [
+			{
+				\"name\": \"NAME\",
+				\"description\": \"Name for the new VM\"
+			},
+			{
+				\"name\": \"MEMORY\",
+				\"value\": \"4096Mi\",
+				\"description\": \"Amount of memory\"
+			},
+			{
+				\"name\": \"CPU_CORES\",
+				\"value\": \"4\",
+				\"description\": \"Amount of cores\"
+			}
+		],
+		\"metadata\": {
+			\"labels\": {
+				\"miq.github.io/kubevirt-is-vm-template\": \"true\",
+				\"kubevirt.io/os\": \"fedora27\"
+			},
+			\"annotations\": {
+				\"description\": \"OCP KubeVirt Fedora 27 VM template\",
+				\"tags\": \"kubevirt,ocp,template,linux\"
+			},
+			\"console.openshift.io/iconClass\": \"icon-fedora\",
+			\"name\": \"fedora-vm-template\"
+		}
+	}"
+
+	Context("with oc command", func() {
+		var vmTemplateJsonFile = ""
+		var vmJsonFile = ""
+
+		BeforeEach(func() {
+			tests.SkipIfNoOc()
+			// login into oc CLI
+			out, err = RunOcCommand("login", "-u", ocUser)
+			Expect(err).ToNot(HaveOccurred())
+			// write testing vm template into file
+			vmTemplateJsonFile, err = WriteJson("testvm-template", vmTemplate)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			if vmTemplateJsonFile != "" {
+				// remove testing vm template
+				err = os.Remove(vmTemplateJsonFile)
+				Expect(err).ToNot(HaveOccurred())
+				vmTemplateJsonFile = ""
+			}
+			if vmJsonFile != "" {
+				// remove testing vm JSON file
+				err = os.Remove(vmJsonFile)
+				Expect(err).ToNot(HaveOccurred())
+				vmJsonFile = ""
+			}
```

Comment:
That is a fair point. It would be indeed much cleaner to add:

```go
AfterEach(func() {
    // logout from oc CLI
    out, err = RunOcCommand("logout")
    Expect(err).ToNot(HaveOccurred())
    ...
}).
```

Reply:
Why do we even have to do this "login"? Can't you just pass in `-server` and `kubeconfig` flags from our code?